CrossDistroLDAPInfra/main.tf

```diff
@@ -0,0 +1,115 @@
+# Input Variables
+variable "access_key" {
+  description = "AWS access key"
+  type        = string
+}
+
+variable "secret_key" {
+  description = "AWS secret key"
+  type        = string
+}
+
+# Provider Configuration
+provider "aws" {
+  region     = "us-east-1"
+  access_key = var.access_key
+  secret_key = var.secret_key
+}
+
+# Common Security Group
+resource "aws_security_group" "allow_ssh" {
+  name_prefix = "allow_ssh"
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+# Debian Instance
```

Comment:
I am wondering if it make sense to have this all in one or keep them in separate files depending on which test we want to run. I don't have a strong opinion yet, but it would be nice to run one environment at a time or all in a GitHub action.